### DISCUSSION TITLE
PnP Search Vertices and Search Box : How to customize more search box and Vertices.. please help

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#2283] [Q&A]
Hi Team, 

We using PnP Modern Search to build as a Enterprise Search solution and I'm looking for help to add more CSS and custom styles to PnP Search Box and PnP Search Vertices but couldn't able to find any way as I'm able to customize the other PnP Search web parts like PnP Search Filters and results.

Please help me or guide us any way , we can customize the PnP Search Box and Vertices in latest PnP Search Modern V4 web part solution? 

Example : Would like to change out of the box search box design with Google search box style, 
                Would like to change Search Vertices background color when before and after mouse over into the vertices to different style. etc.

thank you for help me.

Regards,
gangadhar

**Comment by FranckyC** [comment]
Hi @GangadharMFord, unfortunately, we do not provide custom template capability for search box and search verticals. The only way to customize is either to create your own components and integrate them with search results WP using dynamic data or update the code of existing ones.

**Reply by GangadharMFord** [reply]
thank you Franckyc for clarifying with your response above and as you mentioned , Like we can change search box and vertices functionality with loading sppkg solution into visual studio development and can make changes...?

Question: If in future there was a plan to use newer version of this solution, seems again we need to redevelop or rework on the same changes which are developing from our end or re-deploy the same solution will work?

**Reply by wobba** [reply]
@GangadharMFord you can create your own SPFx solution, like a search box, and make it connect via dynamic data with our web parts.
